### PR TITLE
LUT-26859 Performance enhancement : ManageAppointments.jsp?view=manageAppointments&id_form=XX&context=app

### PR DESCRIPTION
**Pull request by AnasBenjelloun**

### CODE REVIEW

File: src/java/fr/paris/lutece/plugins/appointment/service/AppointmentService.java [hunk 1]
```diff
@@ -291,6 +291,49 @@ public static List<Appointment> findListAppointmentsByFilter( AppointmentFilterD
...
+     * @param appointmentFilter
+     *         the filter
+     * @return a list of appointments
+     */
+    public static List<Integer> findListAppointmentsIdsByFilter( AppointmentFilterDTO appointmentFilter )
+    {
+        return AppointmentHome.findIdsByFilter( appointmentFilter );
+    }
+
+    public static List<AppointmentDTO> findListAppointmentsDTOByFilterByPage( AppointmentFilterDTO appointmentFilter, List<Integer> listIds, int nItemsPerPage,
```
Comment (ryahiaoui) [inline src/java/fr/paris/lutece/plugins/appointment/service/AppointmentService.java:306]: Could you please move this method, either as a private method within the AppointmentJspBean class or into a utility class

File: src/java/fr/paris/lutece/plugins/appointment/web/AppointmentJspBean.java [hunk 2]
```diff
@@ -557,8 +557,14 @@ public synchronized String getManageAppointments( HttpServletRequest request ) t
...
             }
-        List<AppointmentDTO> listAppointmentsDTO = AppointmentService.findListAppointmentsDTOByFilter( _filter );
-
+        _strCurrentPageIndex = AbstractPaginator.getPageIndex( request, AbstractPaginator.PARAMETER_PAGE_INDEX, _strCurrentPageIndex );
+        if ( _strCurrentPageIndex == null )
+        {
+            _strCurrentPageIndex = DEFAULT_CURRENT_PAGE;
+        }
+        _nItemsPerPage = AbstractPaginator.getItemsPerPage( request, AbstractPaginator.PARAMETER_ITEMS_PER_PAGE, _nItemsPerPage, _nDefaultItemsPerPage );
+        List<Integer> listAppointmentsIds = AppointmentService.findListAppointmentsIdsByFilter( _filter );
```
Comment (ryahiaoui) [inline src/java/fr/paris/lutece/plugins/appointment/web/AppointmentJspBean.java:566]: This instruction should be called only in the case where _strCurrentPageIndex is null or equal to DEFAULT_CURRENT_PAGE

**Review by ryahiaoui — COMMENTED** [review]
.

File: src/java/fr/paris/lutece/plugins/appointment/web/AppointmentJspBean.java (hunk 2 above)
Comment (AnasBenjelloun) [reply src/java/fr/paris/lutece/plugins/appointment/web/AppointmentJspBean.java:566]: If I refresh the list only when the table is first displayed (_strCurrentPageIndex is null or equal to DEFAULT_CURRENT_PAGE) I would not be able to display an appointment created after the table was displayed

File: src/java/fr/paris/lutece/plugins/appointment/service/AppointmentService.java (hunk 1 above)
Comment (AnasBenjelloun) [reply src/java/fr/paris/lutece/plugins/appointment/service/AppointmentService.java:306]: the remark concerns the entire findListAppointmentsDTOByFilterByPage method or only the piece of code which calculates the pagination of the IDS?

File: src/java/fr/paris/lutece/plugins/appointment/web/AppointmentJspBean.java (hunk 2 above)
Comment (ryahiaoui) [reply src/java/fr/paris/lutece/plugins/appointment/web/AppointmentJspBean.java:566]: The request is expensive when called on each page. If the user wants to list the appointments created during pagination, they simply need to refresh the filter or go back to the first page.

**Pull request merged**